docs/docs/guides/04_transactions/gas-and-fees.md

```diff
@@ -0,0 +1,83 @@
+
+-   `baseFeePerGas`: base fee from the last block
+-   `gasPrice`: result of the [`eth_gasPrice` RPC method](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gasprice) (legacy purposes only)
+-   `maxPriorityFeePerGas`: result of the [`eth_maxPriorityFeePerGas` RPC method](https://github.com/ethereum/execution-apis/blob/4140e528360fea53c34a766d86a000c6c039100e/src/eth/fee_market.yaml#L29-L42)
+-   `maxFeePerGas`: calculated as `baseFeePerGas * baseFeePerGasFactor + (maxPriorityFeePerGas ?? alternativeMaxPriorityFeePerGas)`
+
+Web3.js transactions may specify `maxFeePerGas` and `maxPriorityFeePerGas` values. If both values are specified, `maxFeePerGas` must be greater than or equal to `maxPriorityFeePerGas`. If `maxFeePerGas` is less than the current base fee, the transaction will not execute until the base fee drops to a value that is less than or equal to the `maxFeePerGas`.
+
+```ts
+const transaction: Transaction = {
+	from: '<SENDER ADDRESS>',
+	to: '<RECEIVER ADDRESS>',
+	value: web3.utils.ethUnitMap.ether,
+};
+
+const feeData: FeeData = await web3.eth.calculateFeeData();
+transaction.maxFeePerGas = feeData.maxFeePerGas;
+transaction.maxPriorityFeePerGas = feeData.maxPriorityFeePerGas;
+```
+
+## Generating Access Lists
+
+An access list specifies the addresses and storage keys that a transaction plans to access. Specifying these elements in advance makes a transaction's gas costs more predictable.
+
+The Ethereum JSON-RPC specifies the [`eth_createAccessList` method](https://github.com/ethereum/execution-apis/blob/4140e528360fea53c34a766d86a000c6c039100e/src/eth/execute.yaml#L54-L97), which accepts a transaction and returns an object that lists the addresses and storage keys that the transaction will access, as well as the approximate gas cost for the transaction if the access list is included. The transaction will not be executed and added to the blockchain. To invoke the `eth_createAccessList` RPC method, use the [`Web3Eth.createAccessList` method](/api/web3-eth/function/createAccessList) and provide the [`TransactionForAccessList`](/api/web3-types/interface/TransactionForAccessList) for which to generate the access list.
+
+Web3.js transactions may specify an access list by providing the [`Transaction.accessList` property](/api/web3/namespace/types#accessList).
+
+```ts
+const transaction: TransactionForAccessList = {
+	from: '<SENDER ADDRESS>',
+	to: '<RECEIVER ADDRESS>',
+	value: web3.utils.ethUnitMap.ether,
+};
+
+const accessListResult: AccessListResult = await web3.eth.createAccessList(transaction);
+transaction.accessList = accessListResult.accessList;
+transaction.gas = accessListResult.gasUsed;
+```
```

Comment:
I’m not sure about the rest of the team’s opinion, but from my experience, mutating data is generally not a good practice and often leads to unexpected outcomes. The example code would be clearer and more reliable if written as suggested below.
 However, feel free to disregard this suggestion if it doesn’t align with the general style of examples in the documentation.

```suggestion
```ts
const transactionDraft: TransactionForAccessList = {
	from: '<SENDER ADDRESS>',
	to: '<RECEIVER ADDRESS>',
	value: web3.utils.ethUnitMap.ether,
};

const accessListResult: AccessListResult = await web3.eth.createAccessList(transactionDraft);

const transaction: TransactionForAccessList = {
	...transactionDraft,
	accessList: accessListResult.accessList,
	gas: accessListResult.gasUsed,
};
```
```